python/arcticdb/version_store/_store.py

```diff
@@ -2076,14 +2077,18 @@ def snapshot(
             Optional list of symbols to be excluded from the snapshot.
         versions: `Optional[Dict[str, int]]`, default=None
             Optional dictionary of versions of the symbols to include in the snapshot.
+        allow_partial_snapshot: bool, default=False
+            Optional flag. When partially valid ``versions`` is given,
```

Comment:
My objection is to the phrase "partially valid", we can't expect a user to infer what that means. It is just about the existence of symbol-version pairs, so say that explicitly.